icu4c/source/test/intltest/loctest.cpp

```diff
@@ -4101,6 +4101,34 @@ LocaleTest::TestCreateUnicodeKeywordSet(void) {
                result.find("ca") != result.end());
     assertTrue("set::find(\"co\")",
                result.find("co") != result.end());
+
+    static const char tag[] = "en-US-u-ca-gregory-x-foo";
+    Locale l2 = Locale::forLanguageTag(tag, status);
+    result.clear();
+    l2.getKeywords<std::string>(
+            std::insert_iterator<decltype(result)>(result, result.begin()),
+            status);
+    status.errIfFailureAndReset("getKeywords \"%s\"", l2.getName());
+    assertTrue("getKeywords set::find(\"calendar\")",
+               result.find("calendar") != result.end());
+    assertTrue("getKeywords set::find(\"ca\")",
+               result.find("ca") == result.end());
+    assertTrue("getKeywords set::find(\"x\")",
+               result.find("x") != result.end());
+    assertTrue("getKeywords set::find(\"co\")",
```

Comment:
The test shouldn't be looking for `"co"` here, there's no such thing in the input language tag so it couldn't possibly show up, even if there were bugs. What the test should be looking for here is `"foo"`, which could possibly show up if some bug got values and keys mixed up.